test/e2e/validation_test.go

```diff
@@ -143,3 +148,218 @@ func testNodeTaint(t *testing.T) {
+}
+
+// ensureTestRunnerRoleBinding ensures the proper RoleBinding exists, a requirement for SSHing into a Windows node
+// noop if the RoleBinding already exists.
+func (tc *testContext) ensureTestRunnerRoleBinding() error {
+	rb := rbac.RoleBinding{
+		TypeMeta:   meta.TypeMeta{},
+		ObjectMeta: meta.ObjectMeta{Name: tc.workloadNamespace},
+		Subjects: []rbac.Subject{{
+			Kind:      "ServiceAccount",
+			APIGroup:  "",
+			Name:      tc.workloadNamespace,
+			Namespace: tc.workloadNamespace,
+		}},
+		RoleRef: rbac.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     tc.workloadNamespace,
+		},
+	}
+	_, err := tc.client.K8s.RbacV1().RoleBindings(tc.workloadNamespace).Create(context.TODO(), &rb, meta.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		return errors.Wrap(err, "unable to create role")
+	}
+	return nil
+}
+
+// sshSetup creates all the Kubernetes resources required to SSH into a Windows node
+func (tc *testContext) sshSetup() error {
+	if err := tc.ensureTestRunnerSA(); err != nil {
+		return errors.Wrap(err, "error ensuring SA created")
+	}
+	if err := tc.ensureTestRunnerRole(); err != nil {
+		return errors.Wrap(err, "error ensuring Role created")
+	}
+	if err := tc.ensureTestRunnerRoleBinding(); err != nil {
+		return errors.Wrap(err, "error ensuring RoleBinding created")
+	}
+	return nil
+}
+
+// copyKeySecret copies the current private-key secret from the WMCO namespace into the testContext's test namespace
+func (tc *testContext) copyKeySecret() error {
+	currentSecret, err := tc.client.K8s.CoreV1().Secrets(tc.namespace).Get(context.TODO(), secrets.PrivateKeySecret, meta.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to get current private key secret from WMCO namespace")
+	}
+
+	// Remove secret from test namespace if it already exists, so the new secret can be created
+	secretClient := tc.client.K8s.CoreV1().Secrets(tc.workloadNamespace)
+	err = secretClient.Delete(context.TODO(), secrets.PrivateKeySecret, meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "unable to delete existing copied secret")
+	}
+	newSecret := core.Secret{
+		ObjectMeta: meta.ObjectMeta{Name: secrets.PrivateKeySecret},
+		Data:       currentSecret.Data,
+	}
+	_, err = secretClient.Create(context.TODO(), &newSecret, meta.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not create secret")
+	}
+	return nil
+}
+
+// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
+// by the provided IP.
+func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+	// Copy the current WMCO private key to the test namespace, so that the SSH pod which will run in the test
+	// namespace will have access to it
+	if err := tc.copyKeySecret(); err != nil {
+		return "", errors.Wrap(err, "error copying private-key secret to test namespace")
+	}
```

Comment:
Shouldn't this also be a one time operation similar to `sshSetup()`?

Reply:
The private key secret changes multiple times throughout the creation suite so when SSHing, the current key has to be copied over and used. It should be possible to pull this out of this function, but for simplicity sake I thought it made sense to keep it in here, since its so subject to change.

Thats also the reasoning behind deleting the existing key and copying over the current one https://github.com/openshift/windows-machine-config-operator/pull/474#discussion_r657406633

Reply:
Then this should be done where the key is being changed and not all the time. Is that really hard to do? We are going to be making a lot of SSH calls and this sort of churn is not required when it can be easily avoided.

Reply:
Moved into the secret creation/deletion function and updated the commit message